### DISCUSSION TITLE
Simulating single compartment neurons?

### DISCUSSION THREAD

**Discussion by ntolley** [Poirazi-Lab/dendrify#38] [Q&A]
I am attempting to build a network with excitatory/inhibitory neuron populations. For the inhibitory neurons I intend to model them as a single compartment. It's unclear to me is the `Soma` class can be used for this, as the examples on the website seem to expect this to be connected in a graph before calling `make_neurongroup()`

If this isn't currently possible I'm happy to give a shot at opening a PR as I'm interested in the project

FYI for the time being I'll use the point neuron models

Thanks!

**Comment by mpgl** [comment]
Dear Nicholas,

Thank you for your question and for using Dendrify.

To simulate point neurons using Dendrify, you can use the [PointNeuronModel](https://dendrify.readthedocs.io/en/latest/api/pointneuronmodel.html) class directly. There are several [examples](https://dendrify.readthedocs.io/en/latest/examples/point.html) in the documentation that demonstrate how to use this class. The Soma class is typically used for  multi-compartmental models, and may not be necessary for your intended single-compartment inhibitory neuron model.

If you have specific reasons for wanting to use the Soma class or if there's something I'm missing, please let me know. I'm happy to help further and would love to hear more about your use case.

Best regards,
Michalis

**Comment by ntolley** [comment]
Thank you for the quick reply! I suppose for my purposes it's not immediately necessary to simulate single compartment models. The main motivation was consistency w.r.t. the level of detail for different types of neurons in the network (i.e. not mixing point neurons with compartmental neurons), but functionally I imagine this has little impact.

For now I'll push forward with using multicompartment excitatory neurons, and point neurons models for inhibitory.
